src/github.com/couchbase/sync_gateway/db/change_listener.go

```diff
@@ -113,7 +113,17 @@ func (listener *changeListener) Wait(keys []string, counter uint64) uint64 {
 		if curCounter != counter {
 			return curCounter
 		}
-		listener.tapNotifier.Wait()
+		//Only wait if the parent changes feed has not been terminated
+		select {
+			case _, more := <-terminator:
+				if more { //Terminator has been closed
+					listener.tapNotifier.Wait()
+				} else {
+					return counter
+				}
+			default:
+				listener.tapNotifier.Wait()
+		}
```

Comment:
I think this would be easier to follow as:
select {
   case <-terminator:
        return counter
   default:
}
listener.tapNotifier.Wait()

Is there a scenario that this wouldn't cover?